Ansible/playbook.yml

```diff
@@ -0,0 +1,136 @@
+      - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCydBZtnehII5hsGLcN7Do6XvBm6tG4n4M4s3HoKRifDIpKEvw+MgZ0nSqexYs6nrN5iBdL0Vf+c3Mv3AYUTFKhDl2JTu1FkDMv0Htkr7lPi4FWKEl4+10cykggBtgJIi3AknSVET8UrWvEAxTYBIqAqTHOa2dylAD2OQHKzB7A6TmRIzNz45uZWkYgKGQDT+ZXVJbrXRurWzhQ6moJe9ETacKwkZfOXzIVehkgE1wtxgB3GygVAarOKfptgRBHlxZCObbBnUSwUPE4ZcvmZ5iRluyqlZxxveMH3Zr9pwFzuLpl278vGDnVRctpAOb1q0e1pU6uRzNmSNDLf5yIEeDf1V2XGHSe8ojjwi+LA1JVkIQYQ0mFgk+F20aSREX1TPIVo8VckHezvqdDC8Uuml6dVemRIB0pdRjk11/disOXLzC7ESDz1ASw5oq/P7Ehi3+CyOhGpkILxf79nog3T3/D0TyefsIoFgZiUfpQsv4axkKJd5vIwKH+o2DpzOh8sWM= vagrant@master"
+
+    new_user: "matusik"
+    new_user_password: "matusik"
+
+  tasks:
+
+    #- name: Test connection
+    #  local_action: command ssh -q -o BatchMode=yes -o ConnectTimeout=3 {{ inventory_hostname }}
+    #  register: test_user
+    #  ignore_errors: true
+    #  changed_when: false
+
+    - name: Update and Upgrade apt packages
+      apt:
+        update_cache: yes
+        upgrade: dist
+
+    - name: Create a non-root user
+      user:
+        name: "{{ new_user }}"
+        password: "{{ new_user_password | password_hash('sha512') }}"
+        groups: sudo
+        shell: /bin/bash
+        state: present
+
+    - name: Create .ssh directory for the new user
+      file:
+        path: "/home/{{ new_user }}/.ssh"
+        state: directory
+        owner: "{{ new_user }}"
+        group: "{{ new_user }}"
+        mode: '0700'
+
+    - name: Add SSH public keys for the new user
+      authorized_key:
+        user: "{{ new_user }}"
+        state: present
+        key: "{{ item }}"
+      loop: "{{ ssh_public_keys }}"
+
+    - name: Disable root login via SSH / Disable remote login for root
+      lineinfile:
+        path: /etc/ssh/sshd_config
+        regexp: '^PermitRootLogin'
+        line: 'PermitRootLogin no'
+        state: present
+          #notify: restart ssh
+
+        #- name: Change SSH port
+        #lineinfile:
+        #path: /etc/ssh/sshd_config
+        #regexp: '^#?Port '
+        #line: 'Port 2222'
+        #notify: restart ssh
+          #
+    - name: Disable password authentication for SSH
+      lineinfile:
+        path: /etc/ssh/sshd_config
+        regexp: '^#?PasswordAuthentication'
+        line: 'PasswordAuthentication no'
+          #notify: restart ssh
+  
+    #- name: Allow specific users for SSH
+    #  lineinfile:
+    #    path: /etc/ssh/sshd_config
+    #    line: 'AllowUsers {{ new_user }}'
+    #  notify: restart ssh
```

Comment:
also a good practice, why not apply it?

Reply:
yes, but repeated launch of the playbook with current user (Vagrant) wont work